ui/app/components/MethodTreeComponent/MethodTreeComponent.js

```diff
@@ -183,45 +185,48 @@ class MethodTreeComponent extends Component {
 
   handleFilterChange (e) {
     const { pathname, query } = this.props.location;
-    this.props.router.push({ pathname, query: { ...query, filterText: e.target.value } });
+    this.props.router.push({ pathname, query: { ...query, [this.props.filterKey]: e.target.value } });
   }
 
   render () {
-    const { filterText } = this.props.location.query;
+    const filterText = this.props.location.query[this.props.filterKey];
     const { nextNodesAccessorField } = this.props;
     let nodeIndexes;
-    if(nextNodesAccessorField === 'parent') {
+    if (nextNodesAccessorField === 'parent') {
       nodeIndexes = this.props.nodeIndexes.map((nodeIndex) => [nodeIndex, undefined]);
-    }else{
+    } else {
       nodeIndexes = this.props.nodeIndexes;
-   }
-
-    const treeNodes = this.getTree(nodeIndexes, '', filterText);
+    }
+    const treeNodes = this.renderSubTree(nodeIndexes, '', 0, false, filterText);
 
     return (
-      <div style={{ padding: '0 10px', margin: '20px 0px' }}>
-        <div>
-          <div>
-            <div style={{ width: '100%', position: 'relative', height: 30 }}>
-              <div className={`${styles.code} ${styles.heading}`}>
-                Method name
-                <input
-                  className={styles.filter}
-                  type="text"
-                  placeholder="Type to filter"
-                  autoFocus
-                  defaultValue={filterText}
-                  onChange={this.debouncedHandleFilterChange}
-                />
-              </div>
-              <div className={`${styles.onStack} ${styles.heading}`}/>
-            </div>
-            {treeNodes}
-          </div>
-          {filterText && !treeNodes.length && (
-            <p className={styles.alert}>Sorry, no results found for your search query!</p>
-          )}
+      <div>
+        <div className={treeStyles.container}>
+          <table>
+            <thead><tr>
+              <th className={treeStyles.fixedRightCol1}>Samples</th>
+              <th>
+                <div className={`mdl-textfield mdl-js-textfield mdl-textfield--floating-label is-dirty is-upgraded ${styles.filterBox}`}>
+                  <input
+                    className={`mdl-textfield__input`}
+                    type="text"
+                    defaultValue={filterText}
+                    autoFocus
+                    onChange={this.debouncedHandleFilterChange}
+                    id="method_filter"
+                  />
+                  <label htmlFor="method_filter" className="mdl-textfield__label">Stack Line</label>
```

Comment:
Stack line may not be that intuitive
![screen shot 2017-04-21 at 7 54 47 pm](https://cloud.githubusercontent.com/assets/3403968/25281888/72add844-26cc-11e7-95fc-c326d88702b3.png)
![screen shot 2017-04-21 at 7 52 56 pm](https://cloud.githubusercontent.com/assets/3403968/25281916/87d6552a-26cc-11e7-8f09-347eb48e1d00.png)